ruby/github_repo_handler.rb

```diff
@@ -17,34 +18,32 @@ def get_account_name
     STDIN.gets.chomp
   end
 
+# **********         Main clone repository logic.       **********
+  def clone_repositories
+    return unless connection? && user_exists?
+
+    calulate_number_of_requests_to_send().times do |index|
+      response_page_logic(index+1)  # starting at one because page=0 and page=1 are identical.
+    end
+  end
+
   def connection?
     result = Net::Ping::HTTP.new('https://api.github.com').ping?
     puts Error.network_error unless result
     result
   end
 
-  def github_url
-    "https://api.github.com/users/#{account_name}"
-  end
-
-  def github_response
-    parse_response(github_url)
-  end
-
   def user_exists?
-    result = github_response['message'].nil?
+    result = github_user_info_request()['message'].nil?
```

Comment:
Why the parens? This is valid but heavily against Ruby conventions if empty.

Reply:
Writing the code this way tells the viewer of the code that `github_user_info_request` needs to finish and then functions always return in ruby so  the message is derived from the function return.

I am willing to change the areas where I implemented these changes. However if there are no parenthesis a variable can either be a function or a variable. 
(more mental overhead to figure out.)

Since this is a program that has a large number of functions use parenthesis and have values passed in to those parenthesis. The code was more uniform if I took the magic out. And had all of the functions behave the same. 

All of the functions implicitly return a value so regardless of whether a developer sees it with parenthesis or not they know it resolves to the returned result.

My favorite use of a function with empty parenthesis in this project :
`github_calulate_number_of_requests_to_send().times `

The code is telling the programmer a behavior, which then derives a result and then we iterate over it that many `.times`. I did not change the normal `.times` behavior.

That was a really hard part of the code to make more readable.

`30.times makes` a lot of sense.
`number_of_pages.times` is difficult to discern what is a page how to we make one?
`github_calulate_number_of_requests_to_send().times ` 
-> lets you know the function derives a number we will do it that many times.

Reply:
@murjax, I think it will be easier if to triage this to a future pull request.
Being that this is a wide sweeping change and I imagine there will be a large amount of discussion around the change.